api/app/controllers/webhook.controller.js

```diff
@@ -0,0 +1,48 @@
+const db = require("../sequelize/models");
+const User = db.user;
+const Review = db.review;
+const Transcript = db.transcript;
+const Op = db.Sequelize.Op;
+
+
+exports.create = async (req, res) => {
+    // const action = req.headers["x-github-event"];
+    const pull_request = req.body;
+     const action = pull_request.action;
+
+
+    // Check if it's a pull request event
+    if (action === 'opened' || action === 'closed' || action === 'synchronize') {
+        const html_url = pull_request.pull_request.html_url;
+
+        console.log(`Received pull request event for PR: ${html_url}`);
+
+        try {
+            // Check if the PR URL exists in the database
+            const existingReview = await Transcript.findOne({ where: { pr_url: html_url } });
+
+            if (existingReview) {
+                // PR is merged, update the mergeAt timestamp
+                existingReview.mergedAt = new Date();
+                existingReview.archivedAt = new Date();
+
+                await existingReview.save();
+
+                console.log(`Review with PR URL ${html_url} updated successfully.`);
+            } else {
+                console.log(`Review with PR URL ${html_url} not found in the database.`);
+            }
+        } catch (error) {
+            console.error('Error occurred while updating the review:', error);
```

Comment:
Do you think we should take another action asides from logging the error to the console?
We should have better error handling for task failure.

Reply:
This is just a catch, not the final